### PR TITLE
[bazel] Add bazel to lists of required packages and to CI

### PR DESCRIPTION
**Pull request by drewmacrae**
Signed-off-by: Drew Macrae <drewmacrae@google.com>

### CODE REVIEW

File: util/container/Dockerfile [hunk 1]
```diff
@@ -46,6 +46,13 @@ RUN OBS_URL="https://download.opensuse.org/repositories"; \
     mv "$TMPDIR/obs.asc"  /etc/apt/trusted.gpg.d/obs.asc && \
     mv "$TMPDIR/obs.list" /etc/apt/sources.list.d/edatools.list
 
+# Add bazel repository to apt sources.
+RUN apt-get install -y apt-transport-https curl gnupg \
+    && curl -fsSL https://bazel.build/bazel-release.pub.gpg | gpg --dearmor > bazel.gpg \
+    && mv bazel.gpg /etc/apt/trusted.gpg.d/ \
+    && echo "deb [arch=amd64] https://storage.googleapis.com/bazel-apt stable jdk1.8" | sudo tee /etc/apt/sources.list.d/bazel.list \
```
Comment (milesdai) [inline util/container/Dockerfile:53]: It looks like this line is breaking in the `build_docker_containers` job. I think Docker doesn't require running with this with `sudo`.

Comment (drewmacrae) [reply util/container/Dockerfile]: Thanks looks like I missed that one.

**Comment by drewmacrae** [comment]
By the way, I couldn't find a pointer to the failure in the azure-pipelines.yml file here and just saw that the CI pipeline didn't start. I found a record of the failure in the queue here: https://dev.azure.com/lowrisc/opentitan/_build?definitionId=9

**Comment by drewmacrae** [comment]
The default pipelines give us a couple CPU hours per check which is just enough to build verilator, so the bazel tests in CI don't build verilator or targets that depend on it right now.

**Review by mcy — APPROVED** [review]

File: ci/install-package-dependencies.sh [hunk 2]
```diff
@@ -86,6 +86,20 @@ echo "$EDATOOLS_REPO" > "$TMPDIR/obs.list"
 sudo mv "$TMPDIR/obs.asc"  /etc/apt/trusted.gpg.d/obs.asc
 sudo mv "$TMPDIR/obs.list" /etc/apt/sources.list.d/edatools.list
 
+# Install gcc-8 and set it as the default
```
Comment (alphan) [inline ci/install-package-dependencies.sh:89]: nit: below commands seem to be installing `gcc-9` while the comment says `gcc-8`.

**Review by alphan — APPROVED** [review]

File: ci/install-package-dependencies.sh (hunk 2 above)
Comment (drewmacrae) [reply ci/install-package-dependencies.sh]: Done

**Review by mundaym — APPROVED** [review]
LGTM, thanks for adding this! Let me know when you're ready for this to be merged.

**Review by milesdai — APPROVED** [review]
LGTM

**Review by arunthomas — APPROVED** [review]
LGTM. Thanks, Drew!

**Comment by alphan** [comment]
@drewmacrae I merged #10280 but it looks like the `pwrmgr_regs` dependencies were missing the "top_earlgrey" folder in their paths. I think it's a good idea to rebase this PR and include that fix as well.

**Comment by drewmacrae** [comment]
> @drewmacrae I merged #10280 but it looks like the `pwrmgr_regs` dependencies were missing the "top_earlgrey" folder in their paths. I think it's a good idea to rebase this PR and include that fix as well.

I have a fix for that in https://github.com/lowRISC/opentitan/pull/10285 now. Not sure why the original issue didn't prevent this PR from passing CI, but I came across the issue in a separate effort to run nightly tests on my workstation. I like PRs to have seperate and separable units of work, and I think that PR stands on its own.

**Comment by drewmacrae** [comment]
Thanks @alphan for pointing out that issue with the pwrmgr_regs fix. Since it's submitted we should be able to merge this one unless something else breaks bazel. My availability is good over the next couple days to monitor CI, so I'd accept any help I can get merging this in.

**Comment by drewmacrae** [comment]
This appears to catch an issue with our bazel config or https://github.com/lowRISC/opentitan/pull/10148 I'm tracking it in slack while we consider reverting this

**Comment by martin-lueker** [comment]
Hi everyone.   This is now impeding our progress at WDC.   CI is failing for us, we don't have bazel installed in our design center and given the nature of our system it may take a while.  Can we please roll this back until can adapt? 

I'm sure this was announced in the Software meeting, but I did not see a broad general announcement for this.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -716,3 +716,19 @@ jobs:
...
+
+- job: bazel_build_and_test
+  displayName: Bazel Build and Test
+  dependsOn: lint
+  condition: and(succeeded(), eq(dependencies.lint.outputs['DetermineBuildType.onlyDocChanges'], '0'))
+  pool:
+    vmImage: ubuntu-18.04
+  steps:
+  - template: ci/install-package-dependencies.yml
+    # The default timeout is too short to build verilator so let's skip it for the first pass
```
Comment (martin-lueker) [inline azure-pipelines.yml:729]: For ease of reproducing this: could we make a script in ci/scripts that does this all?

Comment (martin-lueker) [reply azure-pipelines.yml:729]: Thanks!

**Pull request merged**